tests/ui/drop/lint-tail-expr-drop-order.rs

```diff
@@ -0,0 +1,54 @@
+//@ compile-flags: -Z unstable-options
+//@ edition: 2024
+
+#![deny(tail_expr_drop_order)]
+#![feature(shorter_tail_lifetimes)]
+
+struct LoudDropper;
+impl Drop for LoudDropper {
+    fn drop(&mut self) {
+        println!("loud drop")
+    }
+}
+impl LoudDropper {
+    fn get(&self) -> i32 {
+        0
+    }
+}
+
+fn should_lint() -> i32 {
+    let x = LoudDropper;
+    // Should lint
+    x.get() + LoudDropper.get()
+    //~^ ERROR: these values and local bindings have significant drop implementation that will have a different drop order from that of Edition 2021
+    //~| WARN: this changes meaning in Rust 2024
+}
+
+fn should_lint_closure() -> impl FnOnce() -> i32 {
+    let x = LoudDropper;
+    move || x.get() + LoudDropper.get()
+    //~^ ERROR: these values and local bindings have significant drop implementation that will have a different drop order from that of Edition 2021
+    //~| WARN: this changes meaning in Rust 2024
```

Comment:
Remark: I don't remember if compiletest parsing cares, but maybe

```suggestion
    move || x.get() + LoudDropper.get()
    //~^ ERROR these values and local bindings have significant drop implementation that will have a different drop order from that of Edition 2021
    //~| WARN this changes meaning in Rust 2024
```

Reply:
It turns out that the lint is issued twice. Let me look for a way to issue this just once. 😄

Reply:
oh lol

Reply:
Yeah this duplication is resolved